File: android/app/src/main/kotlin/com/example/fluttersample/data/CarData.kt
package com.example.fluttersample.data

import org.json.JSONObject

data class CarData(
    var carSpeed: Int = 0,
    var currentCarGear: String = "",
    var fuelLevel: Float = 0.0F,
    var fuelLevelLow: Boolean = false,
    var parkingBrake: Boolean = true,
    var hazardLights: Boolean = false,
    var highBeamLights: Boolean = false,
    var headLights: Boolean = false,
) {
    fun toJson(): JSONObject {
        val json = JSONObject()
        json.put("carSpeed", carSpeed)
        json.put("currentCarGear", currentCarGear)
        json.put("fuelLevel", fuelLevel)
        json.put("fuelLevelLow", fuelLevelLow)
        json.put("parkingBrake", parkingBrake)
        json.put("hazardLights", hazardLights)
        json.put("highBeamLights", highBeamLights)
        json.put("headLights", headLights)
        return json
    }
}

File: lib/model/car_data.dart
class CarData {
  int? carSpeed;
  String? currentCarGear;
  double? fuelLevel;
  bool? fuelLevelLow;
  bool? parkingBrake;
  bool? hazardLights;
  bool? highBeamLights;
  bool? headLights;

  CarData(
      {this.carSpeed,
        this.currentCarGear,
        this.fuelLevel,
        this.fuelLevelLow,
        this.parkingBrake,
        this.hazardLights,
        this.highBeamLights,
        this.headLights});

  CarData.fromJson(Map<String, dynamic> json) {
    carSpeed = json['carSpeed'];
    currentCarGear = json['currentCarGear'];
    fuelLevel = json['fuelLevel'];
    fuelLevelLow = json['fuelLevelLow'];
    parkingBrake = json['parkingBrake'];
    hazardLights = json['hazardLights'];
    highBeamLights = json['highBeamLights'];
    headLights = json['headLights'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['carSpeed'] = carSpeed;
    data['currentCarGear'] = currentCarGear;
    data['fuelLevel'] = fuelLevel;
    data['fuelLevelLow'] = fuelLevelLow;
    data['parkingBrake'] = parkingBrake;
    data['hazardLights'] = hazardLights;
    data['highBeamLights'] = highBeamLights;
    data['headLights'] = headLights;
    return data;
  }
}